resolver/src/main/java/io/netty/resolver/DefaultNameResolver.java

```diff
@@ -22,49 +22,39 @@
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.net.UnknownHostException;
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 /**
  * A {@link NameResolver} that resolves an {@link InetSocketAddress} using JDK's built-in domain name lookup mechanism.
  * Note that this resolver performs a blocking name lookup from the caller thread.
  */
-public class DefaultNameResolver extends SimpleNameResolver<InetSocketAddress> {
+public class DefaultNameResolver extends SimpleNameResolver<InetAddress> {
 
     public DefaultNameResolver(EventExecutor executor) {
         super(executor);
     }
 
-    @Override
-    protected boolean doIsResolved(InetSocketAddress address) {
-        return !address.isUnresolved();
+    /**
+     * Creates a new {@link AddressResolver} that will use this name resolver underneath.
+     */
+    public AddressResolver<InetSocketAddress> asAddressResolver() {
+        return new InetSocketAddressResolver(executor(), this);
```

Comment:
Oh right! And as `InetSocketAddressResolver` is very inexpensive, is it ok if I compute it eagerly?